apps/picsa-tools/option-tool/src/app/pages/home/home.component.ts

```diff
@@ -27,13 +28,35 @@ export class HomeComponent implements OnDestroy {
   private editorIndex = 0;
   private componentDestroyed$ = new Subject();
 
+  public status = '';
+  public disabled = false;
```

Comment:
nit(blocking) - I see you have used a `disabled` status but it is not linked to the UI. It would be good to rename as `shareDisabled` and link to the share button like
```html
<button mat-button [disabled]="shareDisabled">
```
This will prevent the user repeatedly trying to share while it is preparing. You could also show the status text within the button instead of the word `Share`